apps/studio/src/features/dashboard/components/DeleteResourceModal/DeleteResourceModal.tsx

```diff
@@ -0,0 +1,150 @@
+import { useState } from "react"
+import {
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react"
+import {
+  Button,
+  Checkbox,
+  ModalCloseButton,
+  useToast,
+} from "@opengovsg/design-system-react"
+import { ResourceType } from "~prisma/generated/generatedEnums"
+import { useAtom } from "jotai"
+import upperFirst from "lodash/upperFirst"
+
+import { isAllowedToHaveChildren } from "~/utils/resources"
+import { trpc } from "~/utils/trpc"
+import {
+  DEFAULT_RESOURCE_MODAL_STATE,
+  deleteResourceModalAtom,
+} from "../../atoms"
+import { DeleteResourceModalState } from "../../types"
+
+const getResourceLabel = (
+  resourceType: ResourceType,
+): Lowercase<Extract<ResourceType, "Page" | "Collection" | "Folder">> => {
+  if (!isAllowedToHaveChildren(resourceType)) {
+    return "page"
+  }
+
+  if (resourceType === ResourceType.Collection) {
+    return "collection"
+  }
+
+  return "folder"
+}
+
+const getWarningText = (resourceType: ResourceType): string => {
+  if (!isAllowedToHaveChildren) {
+    return "This cannot be undone"
+  }
+
+  if (resourceType === ResourceType.Collection) {
+    return "All pages under this collection will be deleted. This cannot be undone."
+  }
+
+  return "All folders and pages under this folder will be deleted. This cannot be undone."
+}
+
+interface DeleteResourceModalProps {
+  siteId: number
+}
+
+export const DeleteResourceModal = ({
+  siteId,
+}: DeleteResourceModalProps): JSX.Element => {
+  const [{ resourceId, ...rest }, setDeleteCollectionModalState] = useAtom(
+    deleteResourceModalAtom,
+  )
+  const onClose = () =>
+    setDeleteCollectionModalState(DEFAULT_RESOURCE_MODAL_STATE)
```

Comment:
nit: as discussed offline, feel like the atom should only consist of the `resourceId` to manage, everything else should be handled by the modal. Having the whole object may add footguns for if certain dependant object props are not passed in at the same time. But for now still ok.